plugins/rapid7_insightvm_cloud/icon_rapid7_insightvm_cloud/actions/asset_search/action.py

```diff
@@ -12,25 +13,27 @@ def __init__(self):
         )
 
     def run(self, params={}):
-        asset_crit = params.get(Input.ASSET_CRITERIA)
-        vuln_crit = params.get(Input.VULN_CRITERIA)
-        size = params.get(Input.SIZE, 200)
-        sort_criteria = params.get(Input.SORT_CRITERIA, dict())
-        parameters = list()
+        body = clean(
+            {"asset": params.pop(Input.ASSET_CRITERIA, None), "vulnerability": params.pop(Input.VULN_CRITERIA, None)}
+        )
+        parameters = {
+            Input.SIZE: self._get_size(params),
+            "currentTime": params.get(Input.CURRENT_TIME),
+            "comparisonTime": params.get(Input.COMPARISON_TIME),
+        }
+        for key, value in params.get("sort_criteria", {}).items():
+            parameters["sort"] = f"{key},{value}"
+
+        parameters = clean(parameters)
+        body = clean(body)
```

Comment:
If `body` is _cleaned_ above then do we need to do it again here?

Reply:
yes, we don't need that, good catch :)